python/ray/data/tests/test_dataset.py

```diff
@@ -1782,7 +1782,7 @@ def test_iter_batches_basic(ray_start_regular_shared):
 
     # blocks format.
     for batch, df in zip(ds.iter_batches(batch_format="native"), dfs):
```

Comment:
Hmm we might want to remove "native" format, since it exposes the implementation detail of the block format.

Reply:
Currently, the `batch_format` parameter is optional. If we remove the "native" option, should we make this parameter required or set "arrow" as the default option?

Reply:
Hmm I think we should probably (1) make the default "pandas", (2) remove "native", and (3) add a way to inspect / cast the format of datasets. But this should probably be a separate PR.

Reply:
Let's do it in a separate PR.